autoPyTorch/api/base_task.py

```diff
@@ -1180,6 +1169,95 @@ def fit(self,
         self._clean_logger()
         return pipeline
 
+    def fit_ensemble(
+        self,
+        ensemble_nbest: Optional[int] = None,
+        ensemble_size: Optional[int] = None,
+        precision: int = 32
+    ) -> 'BaseTask':
+
+        # Make sure that input is valid
+        if self.dataset is None or self.opt_metric is None:
+            raise ValueError("fit_ensemble() can only be called after fit. Please call the "
+                             "estimator search() method prior to fit_ensemble().")
+
+        if self._logger is None:
+            self._logger = self._get_logger(self.dataset.dataset_name)
+
+        # Create a client if needed
+        if self._dask_client is None:
+            self._create_dask_client()
+        else:
+            self._is_dask_client_internally_created = False
+
+        manager = self._fit_ensemble(
+            time_left_for_ensembles=self._time_for_task,
+            optimize_metric=self.opt_metric,
+            precision=precision if precision is not None else self.precision,
+            ensemble_size=ensemble_size if ensemble_size is not None else self.ensemble_size,
+            ensemble_nbest=ensemble_nbest if ensemble_nbest is not None else self.ensemble_nbest,
+        )
+
+        manager.build_ensemble(self._dask_client)
+        future = manager.futures.pop()
+        result = future.result()
+        if result is None:
+            raise ValueError("Error building the ensemble - please check the log file and command "
+                             "line output for error messages.")
+        self.ensemble_performance_history, _, _, _ = result
+
+        self._load_models()
+        self._close_dask_client()
+        return self
+
+    def _fit_ensemble(
+        self,
+        time_left_for_ensembles: float,
+        optimize_metric: str,
+        ensemble_nbest: int,
+        ensemble_size: int,
+        precision: int = 32,
+    ) -> EnsembleBuilderManager:
+
+        assert self._logger is not None, "logger should be initialised to fit ensemble"
+        if self.dataset is None:
+            raise ValueError("ensemble can only be fitted after fit. Please call the "
+                             "estimator search() method prior to fit_ensemble().")
```

Comment:
Can we clarify which search() method?
For example, users would like to know which `class` or `file` to check.

Reply:
sure